projects/gloo/pkg/syncer/envoy_translator_syncer.go

```diff
@@ -142,12 +142,17 @@ func (s *translatorSyncer) syncEnvoy(ctx context.Context, snap *v1snap.ApiSnapsh
 		}
 	}
 
-	allReports.Accept(snap.Upstreams.AsInputResources()...)
-	allReports.Accept(snap.UpstreamGroups.AsInputResources()...)
 	// Only mark non-kube gateways as accepted
 	// Regardless, kube gw proxies are filtered out of these reports before reporting in translator_syncer.go
 	allReports.Accept(nonKubeProxies.AsInputResources()...)
 
+	// mark Upstream[Group]s as Accepted initially, but only if we have at least 1 edge proxy;
```

Comment:
>seems frail to assume certain behavior based on the existence of an edge proxy

I agree with your statement but I actually think it doesn't apply here. This check is effectively saying that this syncer is not going to mark status as Accepted if I'm not also going to translate that same resource. The presence of an edge proxy is just the mechanism it uses to know if it will also do translation on that resource.

>side note: why don't the edge and kube gw translators each read/write their own statuses that don't overlap?

This is probably what we need to do if this stays around long term. We have the namespaced status concept that could be applied to this, although it 1. is a bit a misuse and 2. will tie us further into legacy solo-kit types